### PR TITLE
Write a home page and simple http server

### LINKED ISSUE #42: Write a home page and simple http server
Write a Go server that serves a static index.html file and an API
The API should have one endpoint implemented, which returns a random number between 23 and 50
The html page should have a basic header and body. The body should be responsive and look nice on all screen sizes. Pick some soft colors for the background and content container.
Add a button to the html page, and when clicked, make a request to the "random number" endpoint on the server, and insert the response on the page.

---

files:index.html,main.go

### PR DESCRIPTION
**Pull request by black-crowned-night-heron**
In this task, I've created an html page [index.html] which includes a button to fetch a random number from the server. On the other hand a Go server [main.go] has been created to serve this static file and an API endpoint /get-number to return a random number between 23 and 50. When number gets fetched from server, it is inserted into html page.
For alignment adjusting and responsiveness, CSS Flexbox is being used. The background colors are soft shades of grey. Hope this fits the requirement well.

Resolves #42

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,41 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title>Home Page</title>
+    <style>
```
Comment (mobyvb) [inline index.html:5]: please update with some colors that aren't white or grey

Comment (black-crowned-night-heron) [reply index.html:5]: I have updated the colors as per your request. The body's background color is now purple (`#ab82ff`) and container's background color is now a light orange (`#ffc67d`). The updated colors should make the webpage more vibrant and visually appealing.

File: main.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	"strconv"
+	"time"
+)
+
+func main() {
+	http.Handle("/", http.FileServer(http.Dir("./")))
+
+	http.HandleFunc("/get-number", func(w http.ResponseWriter, r *http.Request) {
+		rand.Seed(time.Now().UnixNano())
+		num := rand.Intn(28) + 23 // to get number between 23 and 50
```
Comment (mobyvb) [inline main.go:16]: could you explain how this works?

Comment (black-crowned-night-heron) [reply main.go:16]: This is the main() function of a simple Go server that serves a web page and provides an API endpoint "/get-number".

Let's break it down: 

- `http.Handle("/", http.FileServer(http.Dir("./")))`: This line serves files from the current directory (where this Go program resides) as a static website.

- The `http.HandleFunc("/get-number", func(w http.ResponseWriter, r *http.Request) {...})` function is an endpoint handler for the "/get-number" URL path.

Inside this handler function:

- `rand.Seed(time.Now().UnixNano())`: this seeds (or initializes) the random number generator using the current time in nanoseconds since 1970.

- `num := rand.Intn(28) + 23`: This generates a random number between 0 and 27 (rand.Intn(28)), then adds 23 to bring the range to between 23 and 50.

- `fmt.Fprintf(w, `{"number": %s}`, strconv.Itoa(num))`: Writes the generated number as part of a JSON string to the response.

Finally, `http.ListenAndServe(":3000", nil)` starts the server and listens for requests on port 3000.

Basically, this server, when running, allows you to visit it at localhost:3000 in a web browser, and if you visit localhost:3000/get-number, it will return a JSON string with a random number between 23 and 50.
